teb_local_planner/src/teb_local_planner_ros.cpp

```diff
@@ -721,9 +721,8 @@ bool TebLocalPlannerROS::transformGlobalPlan(const std::vector<geometry_msgs::ms
 
     // get plan_to_global_transform from plan frame to global_frame
     geometry_msgs::msg::TransformStamped plan_to_global_transform = tf_->lookupTransform(
-                global_frame, tf2_ros::fromMsg(plan_pose.header.stamp),
-                plan_pose.header.frame_id, tf2::timeFromSec(0),
-                plan_pose.header.frame_id, tf2::durationFromSec(0.5));
```

Comment:
If the global_plan is not updated then plan_pose.header.stamp is not updated, and since the TF buffer has a fixed cache time (5-10s), then this throws an Extrapolation error in the past.

The current approach gets the global_frame, which is odom, at the timestamp of the plan creation, however if the AMR is relocalized and odom changes, I think that this new odom should be used.

The fix is to simply get the latest odom->map (global_frame->plan_pose.header.frame_id) TF.